docs/transparent-validation/sonoff/sonoff_preparation.md

```diff
@@ -1,26 +1,99 @@
 
 # Sonoff preparation
 
-The below section of the documentation is based on the
-[EPSHome documentation](https://esphome.io/devices/sonoff_s20.html)
+The following page is based on the
+[EPSHome](https://esphome.io/devices/sonoff_s20.html)
+and [Tasmota](https://tasmota.github.io/docs/devices/Sonoff-S26-Smart-Socket)
+documentations.
 
-> The following procedure is for Sonoff S20. It has also been tested on
-  **Sonoff S26 Release 2**, however, the connection method for flashing is
-  different and is described [here](https://tasmota.github.io/docs/devices/Sonoff-S26-Smart-Socket/#s26-release-2).
+## Firmware setup
 
-1. Open the Sonoff case.
+### Connecting the programmer
 
-1. Check if the headers are soldered to the board. If not - 4 pin header
-    raster 2.54 mm should be soldered to the board in accordance with the below
-    images.
+=== "Sonoff S20"
 
-    ![sonoff_without_header](images/sonoff_without_header.jpg)
+    1. Open the Sonoff case.
 
-    ![sonoff_with_header](images/sonoff_with_header.jpg)
+    1. Check if the headers are soldered to the board. If not - 4 pin header
+        raster 2.54 mm should be soldered to the board in accordance with the below
+        images.
 
-1. Connect the USB-UART converter to the USB port.
+        ![sonoff_without_header](images/sonoff_without_header.jpg)
 
-    ![uart connection](images/sonoff-s20-uart.jpg)
+        ![sonoff_with_header](images/sonoff_with_header.jpg)
+
+    1. Connect the USB-UART converter to the USB port.
+
+        ![uart connection](images/sonoff-s20-uart.jpg)
+
+=== "Sonoff S26 R2"
+
+    Please attach the programmer wires as specified in
+    [this step](https://tasmota.github.io/docs/devices/Sonoff-S26-Smart-Socket/#s26-release-2)
+    of Tasmota documentation.
+
+After connecting the wires, remember to plug in the programmer while holding
```

Comment:
We could use a note on what programmer to use. I assume any USB -> UART adapter will work? I'm testing with a CH341a in UART mode since that's what I have on hand.

Reply:
I had used an adapter, so I've now added a [note](https://github.com/Dasharo/docs/pull/748/files#diff-7727a79ac37ea841d98914ee3913aae9b908b884270b313ef49b2c46c3ac03deR13) stating that both a generic adapter and ch341 work